### ISSUE TITLE
Failed to Pull the backend - \auth\parameters.json' does not exist

### ISSUE THREAD

**Issue by KuzonFyre** [aws-amplify/amplify-cli#13667]
### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

18.18.2

### Amplify CLI Version

12.10.1

### What operating system are you using?

Windows

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

No manual changes made. 

### Describe the bug

Similar to this issue #12766 link [here](https://github.com/aws-amplify/amplify-cli/issues/12766). I am unable to push or pull withou t an error, Failed to Pull the backend. I ran this command with --appId and --envName dev. I also ran this with the --debug flag which produces this error ```\amplify\backend\auth\{APPID]\parameters.json' does not exist``` While making this issue I tried the ```amplify diagnose --send-report``` command which gave me ```\amplify\backend\amplify-meta.json' does not exist```. 

### Expected behavior

I can still run my application. All GraphQL endpoints are working normally along with auth. I need to add social providers to my auth setup and I am assuming I can't do that until I have the amplify-meta.json file. 

### Reproduction steps

I can't provide the repo but can provide the local report that was generated. 

### Log output

<details>

```
# Put your logs below this line
PullBackendFault: Failed to pull the backend.
    at attachBackend (C:\snapshot\amplify-cli\build\node_modules\@aws-amplify\cl
                                                                               li-internal\lib\attach-backend.js:55:15)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5
                                                                               5)
    at async Object.run (C:\snapshot\amplify-cli\build\node_modules\@aws-amplify
                                                                               y\cli-internal\lib\commands\pull.js:64:9)
    at async Object.executeAmplifyCommand (C:\snapshot\amplify-cli\build\node_mo
                                                                               odules\@aws-amplify\cli-internal\lib\index.js:194:9)
    at async executePluginModuleCommand (C:\snapshot\amplify-cli\build\node_modu
                                                                               ules\@aws-amplify\cli-internal\lib\execution-manager.js:139:5)
    at async executeCommand (C:\snapshot\amplify-cli\build\node_modules\@aws-amp
                                                                               plify\cli-internal\lib\execution-manager.js:37:9)
    at async Object.run (C:\snapshot\amplify-cli\build\node_modules\@aws-amplify
                                                                               y\cli-internal\lib\index.js:121:5)

File at path: 'C:\Users\NAME\APPNAME\amplify\backend\auth\APPID\parameter
                                                                               rs.json' does not exist
Error: File at path: 'C:\Users\NAME\APPNAME\amplify\backend\auth\APPID\pa
                                                                               arameters.json' does not exist
    at JSONUtilities.readJson (C:\snapshot\amplify-cli\build\node_modules\@aws-a
                                                                               amplify\amplify-cli-core\lib\jsonUtilities.js:44:19)
    at StateManager.getData (C:\snapshot\amplify-cli\build\node_modules\@aws-amp
                                                                               plify\amplify-cli-core\lib\state-manager\stateManager.js:327:56)
    at StateManager.getResourceParametersJson (C:\snapshot\amplify-cli\build\nod
                                                                               de_modules\@aws-amplify\amplify-cli-core\lib\state-manager\stateManager.js:149:25
                                                                               5)
    at ensureAmplifyMetaFrontendConfig (C:\snapshot\amplify-cli\build\node_modul
                                                                               les\@aws-amplify\cli-internal\lib\extensions\amplify-helpers\on-category-outputs-
                                                                               -change.js:105:60)
    at AmplifyToolkit.onCategoryOutputsChange (C:\snapshot\amplify-cli\build\nod
                                                                               de_modules\@aws-amplify\cli-internal\lib\extensions\amplify-helpers\on-category-o
                                                                               outputs-change.js:46:53)
    at initializeEnv (C:\snapshot\amplify-cli\build\node_modules\@aws-amplify\cl
                                                                               li-internal\lib\initialize-env.js:143:31)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5
                                                                               5)
    at async onSuccess (C:\snapshot\amplify-cli\build\node_modules\@aws-amplify\
                                                                               \cli-internal\lib\attach-backend.js:81:17)
    at async attachBackend (C:\snapshot\amplify-cli\build\node_modules\@aws-ampl
                                                                               lify\cli-internal\lib\attach-backend.js:50:9)
    at async Object.run (C:\snapshot\amplify-cli\build\node_modules\@aws-amplify
                                                                               y\cli-internal\lib\commands\pull.js:64:9)
    at async Object.executeAmplifyCommand (C:\snapshot\amplify-cli\build\node_mo
                                                                               odules\@aws-amplify\cli-internal\lib\index.js:194:9)
    at async executePluginModuleCommand (C:\snapshot\amplify-cli\build\node_modu
                                                                               ules\@aws-amplify\cli-internal\lib\execution-manager.js:139:5)
    at async executeCommand (C:\snapshot\amplify-cli\build\node_modules\@aws-amp
                                                                               plify\cli-internal\lib\execution-manager.js:37:9)
    at async Object.run (C:\snapshot\amplify-cli\build\node_modules\@aws-amplify
                                                                               y\cli-internal\lib\index.js:121:5)


```

</details>

### Before submitting, please confirm:

- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

**Comment by ykethan** [comment]
Hey @KuzonFyre, thank you for reaching out. Are you running the pull  with `--appId` and `--envName dev` in an existing project? Does the error occur if you pull the backend into a empty folder outside the existing project directory?

Could download the `#current-cloud-backend.zip` from the deployment bucket(name ending with -deployment) on the S3 console, under the `/auth/<auth-name>/build` do you observe a `parameters.json` file?

**Comment by KuzonFyre** [comment]
@ykethan Thanks for the reply. I run this in the root directory of an exisiting project. I tried running this in an empty directory outside of my project and recived the same error ```\amplify\backend\auth\{APPID]\parameters.json' does not exist```. I do see a parameters.json file in the ```#current-cloud-backend.zip```. Can I copy that into my local?

**Comment by KuzonFyre** [comment]
I tried copying that file into ```\amplify\backend\auth\{APPID]\parameters.json```. I am noticing that the auth name that I have locally is different then the one in the error message.

**Comment by ykethan** [comment]
@KuzonFyre was the auth resource migrated from an older of version Amplify CLI?
Could you provide us the Amplify CLI version being utilized by running `amplify --version`.

**Comment by KuzonFyre** [comment]
The version is 12.10.1

**Comment by ykethan** [comment]
@KuzonFyre From Amplify CLI version 7, the CLI expects a `cli-inputs.json` file in the Amplify auth resource. On a push, the `cli-inputs.json` is transformed into `parameters.json` under the `auth/build` folder. On a pull the cli checks if the `cli-inputs.json` is present. 
Could you provide us the full error message with the path as the path provided `\amplify\backend\auth\APPID\parameters.json` indicates a older CLI version being utilized?

**Comment by KuzonFyre** [comment]
@ykethan My auth folder has the same ID that I expected to see from AWS. That ID is `napkyff72303c/`. This auth folder has a cli-inputs.json file. Note that the error message below, gives me a different auth folder, `napkyb8f78fcb`. The APPID thing I did was just to remove that info because I thought it was potentially sensitive. I didn't ever change my cli version.
```
🛑 Failed to pull the backend.

Learn more at: https://docs.amplify.aws/cli/project/troubleshooting/

PullBackendFault: Failed to pull the backend.
    at attachBackend (C:\snapshot\amplify-cli\build\node_modules\@aws-amplify\cl
                                                                               li-internal\lib\attach-backend.js:55:15)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5
                                                                               5)
    at async Object.run (C:\snapshot\amplify-cli\build\node_modules\@aws-amplify
                                                                               y\cli-internal\lib\commands\pull.js:64:9)
    at async Object.executeAmplifyCommand (C:\snapshot\amplify-cli\build\node_mo
                                                                               odules\@aws-amplify\cli-internal\lib\index.js:194:9)
    at async executePluginModuleCommand (C:\snapshot\amplify-cli\build\node_modu
                                                                               ules\@aws-amplify\cli-internal\lib\execution-manager.js:139:5)
    at async executeCommand (C:\snapshot\amplify-cli\build\node_modules\@aws-amp
                                                                               plify\cli-internal\lib\execution-manager.js:37:9)
    at async Object.run (C:\snapshot\amplify-cli\build\node_modules\@aws-amplify
                                                                               y\cli-internal\lib\index.js:121:5)

File at path: 'C:\Users\hamil\Napky\amplify\backend\auth\napkyb8f78fcb\parameter
                                                                               rs.json' does not exist
Error: File at path: 'C:\Users\hamil\Napky\amplify\backend\auth\napkyb8f78fcb\pa
                                                                               arameters.json' does not exist
    at JSONUtilities.readJson (C:\snapshot\amplify-cli\build\node_modules\@aws-a
                                                                               amplify\amplify-cli-core\lib\jsonUtilities.js:44:19)
    at StateManager.getData (C:\snapshot\amplify-cli\build\node_modules\@aws-amp
                                                                               plify\amplify-cli-core\lib\state-manager\stateManager.js:327:56)
    at StateManager.getResourceParametersJson (C:\snapshot\amplify-cli\build\nod
                                                                               de_modules\@aws-amplify\amplify-cli-core\lib\state-manager\stateManager.js:149:25
                                                                               5)
    at ensureAmplifyMetaFrontendConfig (C:\snapshot\amplify-cli\build\node_modul
                                                                               les\@aws-amplify\cli-internal\lib\extensions\amplify-helpers\on-category-outputs-
                                                                               -change.js:105:60)
    at AmplifyToolkit.onCategoryOutputsChange (C:\snapshot\amplify-cli\build\nod
                                                                               de_modules\@aws-amplify\cli-internal\lib\extensions\amplify-helpers\on-category-o
                                                                               outputs-change.js:46:53)
    at initializeEnv (C:\snapshot\amplify-cli\build\node_modules\@aws-amplify\cl
                                                                               li-internal\lib\initialize-env.js:143:31)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5
                                                                               5)
    at async onSuccess (C:\snapshot\amplify-cli\build\node_modules\@aws-amplify\
                                                                               \cli-internal\lib\attach-backend.js:81:17)
    at async attachBackend (C:\snapshot\amplify-cli\build\node_modules\@aws-ampl
                                                                               lify\cli-internal\lib\attach-backend.js:50:9)
    at async Object.run (C:\snapshot\amplify-cli\build\node_modules\@aws-amplify
                                                                               y\cli-internal\lib\commands\pull.js:64:9)
    at async Object.executeAmplifyCommand (C:\snapshot\amplify-cli\build\node_mo
                                                                               odules\@aws-amplify\cli-internal\lib\index.js:194:9)
    at async executePluginModuleCommand (C:\snapshot\amplify-cli\build\node_modu
                                                                               ules\@aws-amplify\cli-internal\lib\execution-manager.js:139:5)
    at async executeCommand (C:\snapshot\amplify-cli\build\node_modules\@aws-amp
                                                                               plify\cli-internal\lib\execution-manager.js:37:9)
    at async Object.run (C:\snapshot\amplify-cli\build\node_modules\@aws-amplify
                                                                               y\cli-internal\lib\index.js:121:5)
```

**Comment by KuzonFyre** [comment]
Figured it out! So I copied the `cli-inputs.json` from the auth setup that I know is on AWS, `napkyff72303c/` and put the file into a new directory with the name of the folder the error message said it was missing `napkyb8f78fcb`. That pulled sucesfully. I pulled again and it removed the bad auth folder. Upon deploying it referenced `napkyb8f78fcb`. I searched for references for `napkyb8f78fcb` in my files and found a reference in `backend-config.json` file and one in `team-provider-info.json`. I removed those and it worked. Thanks for the help!
